src/main/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepository.java

```diff
@@ -97,7 +91,7 @@ public SimpleReactiveCouchbaseRepository(CouchbaseEntityInformation<T, String> e
 	@Override
 	public <S extends T> Flux<S> saveAll(Iterable<S> entities) {
 		Assert.notNull(entities, "The given Iterable of entities must not be null!");
-		return Flux.fromIterable(entities).flatMap(this::save);
+		return Flux.fromIterable(entities).flatMap(e -> save(e));
 	}
```

Comment:
This call - where one reactive method calls another reactive method - is what causes the problem when the methodMetadata and methodInvocation are unbound/reset.   The "unbind/unset" occurs *before* save(e) is called, therefore in save(e) the threadLocal metadata is not accessible. The suggestion provided by Mark was to capture the threadLocal metadata in the outer reactive method, then passing it as an arg to the inner reactive method (below). Which requires a new method with that signature. It *seems* like not unbinding/resetting is ok as all our tests pass.  So maybe we can leave it as it is without adding methods with extra args.   Another solution might be to change all the methods like this so there is not a reactive method calling a reactive method.

`public <S extends T> Flux<S> saveAll(Iterable<S> entities) {
		Assert.notNull(entities, "The given Iterable of entities must not be null!");
                 PseudoArgs pArgs = template.getPseudoArgs();
		return Flux.fromIterable(entities).flatMap(e -> save(e, pArgs));
	}
`